packages/meteor-wallet/src/lib/meteor-wallet.ts

```diff
@@ -0,0 +1,223 @@
+          );
+        }
+
+        const block = await provider.block({ finality: "final" });
+
+        return nearTransactions.createTransaction(
+          account.accountId,
+          utils.PublicKey.from(accessKey.public_key),
+          transaction.receiverId,
+          accessKey.access_key.nonce + index + 1,
+          actions,
+          utils.serialize.base_decode(block.header.hash)
+        );
+      })
+    );
+  };
+
+  return {
+    async signIn({ contractId, methodNames }) {
+      logger.log("MeteorWallet:signIn", {
+        contractId,
+        methodNames,
+      });
+
+      let signInRequest: IOMeteorWalletSdk_RequestSignIn_Inputs;
+
+      if (notNullEmptyArray(methodNames)) {
+        signInRequest = {
+          methods: methodNames,
+          type: EMeteorWalletSignInType.SELECTED_METHODS,
+          contract_id: contractId,
+        } as IOMeteorWalletSdk_RequestSignIn_Inputs;
+      } else {
+        signInRequest = {
+          type: EMeteorWalletSignInType.ALL_METHODS,
+          contract_id: contractId,
+        } as IOMeteorWalletSdk_RequestSignIn_Inputs;
+      }
+
+      await _state.wallet.requestSignIn(signInRequest);
+      return getAccounts();
+    },
+
+    async signOut() {
+      if (_state.wallet.isSignedIn()) {
+        _state.wallet.signOut();
+      }
+
+      cleanup();
+    },
+
+    async isSignedIn() {
+      if (!_state.wallet) {
+        return false;
+      }
+
+      return _state.wallet.isSignedIn();
+    },
+
+    async getAccounts() {
+      return getAccounts();
+    },
+
+    async signAndSendTransaction({ signerId, receiverId, actions }) {
+      logger.log("MeteorWallet:signAndSendTransaction", {
+        signerId,
+        receiverId,
+        actions,
+      });
+
+      const { contract } = store.getState();
+
+      if (!_state.wallet.isSignedIn()) {
+        throw new Error("Wallet not signed in");
+      }
+
+      if (!receiverId && !contract) {
+        throw new Error("No receiver found to send the transaction to");
+      }
+
+      const account = _state.wallet.account()!;
+
+      return account["signAndSendTransaction_direct"]({
+        receiverId: receiverId ?? contract!.contractId,
+        actions: actions.map((action) => createAction(action)),
+      });
+    },
+
+    async signAndSendTransactions({ transactions }) {
+      logger.log("MeteorWallet:signAndSendTransactions", {
+        transactions,
+      });
+
+      if (!_state.wallet.isSignedIn()) {
+        throw new Error("Wallet not signed in");
+      }
+
+      return _state.wallet.requestSignTransactions({
+        transactions: await transformTransactions(transactions),
+      });
+    },
+  };
+};
+
+export function setupMeteor({
+  iconUrl = "./assets/meteor-icon.png",
+}: MeteorWalletParams_Injected = {}): WalletModuleFactory<InjectedWallet> {
+  return async () => {
+    return {
+      id: "meteor-wallet-injected",
+      type: "injected",
+      metadata: {
+        available: true,
```

Comment:
How does the SDK work? Is it simply a wrapper around `near-api-js` or does it rely on a `window` variable?

I guess the main question is: is this wallet always available as this suggests (including mobile)?

Reply:
yep, we've tested on mobile as well- it should be always available using the method we are using (postMessage communication between windows). On mobile a new tab is opened where the user signs things, which communicates using our SDK asynchronously as well, and closes when done.

It does rely on the `window` variable for its communication, but it doesn't do full redirect / callbacks.